union.go

```diff
@@ -0,0 +1,58 @@
+package rwvfs
+
+import (
+	"io"
+	pathpkg "path"
+
+	"golang.org/x/tools/godoc/vfs"
+)
+
+// Union returns a new FileSystem which is the union of the provided file systems.
+func Union(fileSystems ...FileSystem) FileSystem {
+	if len(fileSystems) == 0 {
```

Comment:
👍🏾

Sent from my iPhone

> On Jun 6, 2015, at 12:45, Richard Musiol <notifications@github.com> wrote:
> 
> In union.go:
> 
> > @@ -0,0 +1,58 @@
> > +package rwvfs
> > +
> > +import (
> > +	"io"
> > +	pathpkg "path"
> > +
> > +	"golang.org/x/tools/godoc/vfs"
> > +)
> > +
> > +// Union returns a new FileSystem which is the union of the provided file systems.
> > +func Union(fileSystems ...FileSystem) FileSystem {
> > +	if len(fileSystems) == 0 {
> For me a switch means "code paths of equal importance", often seen with enum constants. Here we have two special cases that are not related to each other. You may even remove the check for 1 and it would still work fine. That's why I intentionally prefer two if statements.
> 
> —
> Reply to this email directly or view it on GitHub.
>